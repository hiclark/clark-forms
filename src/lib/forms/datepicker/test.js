import React from 'react';

import DatePicker, { renderDatePicker } from './';

describe('renderDatePicker', () => {
  it('renders correctly', () => {
    const props = { input: {}, meta: { touched: false, error: false } };
    expect(shallow(renderDatePicker(props))).toMatchSnapshot();
  });
});

describe('<DatePicker />', () => {
  let wrapper;
  beforeEach(() => {
    const props = {
      columns: 2,
      onCheckboxClick: jest.fn(),
      name: 'Name',
      label: 'Label',
      value: false,
    };
    wrapper = shallow(<DatePicker {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
