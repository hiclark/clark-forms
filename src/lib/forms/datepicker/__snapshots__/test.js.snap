// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker /> renders correctly 1`] = `
<datepicker__Container>
  <datepicker__Label
    htmlFor="Name"
  >
    Label
  </datepicker__Label>
  <Field
    columns={2}
    component={[Function]}
    name="Name"
  />
</datepicker__Container>
`;

exports[`renderDatePicker renders correctly 1`] = `
<div
  onBlur={[Function]}
>
  <DatePicker
    allowSameDay={false}
    dateForm="MM/DD/YYYY"
    dateFormat="L"
    dateFormatCalendar="MMMM YYYY"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    placeholderText="MM/DD/YYYY"
    preventOpenOnFocus={false}
    selected={null}
    shouldCloseOnSelect={true}
    showTimeSelect={false}
    timeCaption="Time"
    timeIntervals={30}
    withPortal={false}
  />
  <Error
    error={false}
    touched={false}
  />
</div>
`;
